serverless/architecture/serverless-serving-architecture.adoc

```diff
@@ -0,0 +1,27 @@
+include::modules/serverless-document-attributes.adoc[]
+[id="serverless-serving-architecture"]
+= Knative Serving architecture
+:context: serverless-serving-architecture
+include::modules/common-attributes.adoc[]
+
+toc::[]
+
+Knative Serving supports deploying and managing serverless applications, by providing a set of objects as Kubernetes custom resources (CRs) that define and control the behavior of serverless workloads on an {product-title} cluster.
+
+These CRs are building blocks to address complex use cases, for example:
+
+* Rapidly deploying serverless containers.
+* Automatically scaling pods.
+// TODO: Add other use cases, more advanced tutorials?
+
+[id="serverless-serving-architecture-custom-resources"]
+== Knative Serving custom resources
+// TODO: Request a good Serving arch diagram
+
+Configuration:: The `configuration.serving.knative.dev` resource maintains the desired state for your deployment. It provides a clean separation between code and configuration. Modifying a configuration creates a new Revision.
+
+Revision:: The `revision.serving.knative.dev` resource is a point-in-time snapshot of the code and configuration for each modification made to the workload. Revisions are immutable objects and can be retained for as long as necessary.
+
+Service:: The `service.serving.knative.dev` resource automatically manages the life cycle of your workload. It controls the creation of other objects to ensure that your application has a Route, a Configuration, and a new Revision for each change to the Service. Services enable traffic routing to your chosen Revision.
+
+Route:: The `route.serving.knative.dev` resource maps a network endpoint to one or more Revisions. You can manage the traffic in several ways, including fractional traffic and named routes.
```

Comment:
I'd reorder these a bit: Service usually comes first and is the most commonly used CRD of them all.

Reply:
So, they're definitively CRDs, not CRs? @markusthoemmes 
https://github.com/openshift/openshift-docs/pull/23920#issuecomment-663693957

Reply:
Yes they are CRDs. The objects you create are CRs. It's a bit of a philosophical question as to how we refer to them here. CRD is the thing that enables the API, CR is the concrete thing the users create.

Reply:
In theory, a Serving **CR** will be responsible for creating the child resources of course.

Reply:
@jboxman confirmed this with Markus in Slack today, so I will change it back to CRDs.

Reply:
@markusthoemmes I think some follow up work to make this clearer in our docs is probably a good idea, also maybe for upstream if its not explained well there because it's a bit confusing in relation to Knative. I don't really want to delay this PR on it since it's blocking other work though and was just planned to be mostly structural changes, so I'll change them back to CRDs for now, and later I'll make sure the concept of CRDs vs CRs is explained and open an issue to track the work on this separately.